test/ExternalGenericMetadataBuilder/Inputs/json2c.swift

```diff
@@ -0,0 +1,143 @@
+// A simple program that takes the JSON output of libswiftGenericMetadataBuilder
+// and converts it into C code that can be bulit as a library.
+import Foundation
+
+enum ParseError: Error {
+  case oddLengthHexString
+  case invalidHex(Substring)
+}
+
+struct Fixup: Decodable {
+  var target: String
+  var addend: Int64
+}
+
+enum AtomContent: Decodable {
+  case bytes([UInt8])
+  case pointer(Fixup)
+
+  static func getString(decoder: Decoder) -> String? {
+    do {
+      return try decoder.singleValueContainer().decode(String.self)
+    } catch {
+      return nil
+    }
+  }
+
+  init(from decoder: Decoder) throws {
+    if let string = Self.getString(decoder: decoder) {
+      var bytes: [UInt8] = []
+      var toParse = Substring(string)
+      while !toParse.isEmpty {
+        let prefix = toParse.prefix(2)
+        if prefix.count == 1 {
+          throw ParseError.oddLengthHexString
+        }
+        guard let byte = UInt8(prefix, radix: 16) else {
+          throw ParseError.invalidHex(prefix)
+        }
+        bytes.append(byte)
+        toParse = toParse.dropFirst(2)
+      }
+      self = .bytes(bytes)
+    } else {
+      let fixup = try Fixup(from: decoder)
+      self = .pointer(fixup)
+    }
+  }
+}
+
+struct Atom: Decodable {
+  var name: String
+  var contents: [AtomContent]
+}
+
+struct Document: Decodable {
+  var atoms: [Atom]
+}
+
+func deUnderscore(_ str: String) -> Substring {
```

Comment:
I find this function name strangely pleasing. `deUnderscore()` seems like it should have a counterpart `unDederscore()` :-)

Reply:
I'll have to put that in the next version.